src/cli/update.rs

```diff
@@ -0,0 +1,386 @@
+
+/// Update dependencies as recorded in the local lock file
+#[derive(Parser, Debug, Default)]
+pub struct Args {
+    #[clap(flatten)]
+    pub config: ConfigCli,
+
+    /// The path to 'pixi.toml' or 'pyproject.toml'
+    #[arg(long)]
+    pub manifest_path: Option<PathBuf>,
+
+    /// Don't modify the environment, only modify the lock-file.
+    #[arg(long)]
+    pub no_install: bool,
+
+    /// Don't actually write the lockfile or update any environment.
+    #[clap(short = 'n', long)]
+    pub dry_run: bool,
+
+    #[clap(flatten)]
+    pub specs: UpdateSpecsArgs,
+}
+
+#[derive(Parser, Debug, Default)]
+pub struct UpdateSpecsArgs {
+    /// The packages to update
+    pub packages: Option<Vec<String>>,
+
+    /// The environments to update. If none is specified, all environments are
+    /// updated.
+    #[clap(long = "env", short = 'e')]
+    pub environments: Option<Vec<String>>,
+
+    /// The platforms to update. If none is specified, all platforms are
+    /// updated.
+    #[clap(long = "platform", short = 'p')]
+    pub platforms: Option<Vec<Platform>>,
+}
+
+/// A distilled version of `UpdateSpecsArgs`.
+struct UpdateSpecs {
+    packages: Option<HashSet<String>>,
+    environments: Option<HashSet<String>>,
+    platforms: Option<HashSet<Platform>>,
+}
+
+impl UpdateSpecs {
+    fn from_args(args: UpdateSpecsArgs) -> Self {
+        Self {
+            packages: args.packages.map(|args| args.into_iter().collect()),
+            environments: args.environments.map(|args| args.into_iter().collect()),
+            platforms: args.platforms.map(|args| args.into_iter().collect()),
+        }
+    }
+
+    /// Returns true if the package should be relaxed.
+    fn should_relax(&self, environment_name: &str, platform: Platform, package: &Package) -> bool {
+        // Check if the platform is in the list of platforms to update.
+        if let Some(platforms) = &self.platforms {
+            if !platforms.contains(&platform) {
+                return false;
+            }
+        }
+
+        // Check if the environmtent is in the list of environments to update.
+        if let Some(environments) = &self.environments {
+            if !environments.contains(environment_name) {
+                return false;
+            }
+        }
+
+        // Check if the package is in the list of packages to update.
+        if let Some(packages) = &self.packages {
+            if !packages.contains(&*package.name()) {
+                return false;
+            }
+        }
+
+        true
+    }
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    let config = args.config;
+    let project =
+        Project::load_or_else_discover(args.manifest_path.as_deref())?.with_cli_config(config);
+
+    // Load the current lock-file, if any. If none is found, a dummy lock-file is
+    // returned.
+    let loaded_lock_file = load_lock_file(&project).await?;
+
```

Comment:
I think we should check the given packages for existence in the lock file, as I now get this:
```
➜ pixi update annotate-types
✔ Lock-file is up-to-date
```
Which initially confused me as i was trying to update `annotated-types` but I had a typo. So I though I didn't understand how the update function worked.

Reply:
Something like?

![image](https://github.com/prefix-dev/pixi/assets/4995967/da76f740-3de7-4fa2-a6b4-42e13f31d8da)

Reply:
The help would be a plus!!!